packages/flutter/lib/src/material/switch.dart

```diff
@@ -558,9 +556,33 @@ class Switch extends StatelessWidget {
   final bool autofocus;
 
   Size _getSwitchSize(BuildContext context) {
-    final ThemeData theme = Theme.of(context);
-    final SwitchThemeData switchTheme = SwitchTheme.of(context);
+    ThemeData theme = Theme.of(context);
+    SwitchThemeData switchTheme = SwitchTheme.of(context);
```

Comment:
This doesn't look right. The default adaptation should handle the theme.platform case analysis and we shouldn't be using ThemeData.copyWith ...

Reply:
Oh thanks for pointing this out! Just updated this part and moved the platform check to `ThemeData.adaptive()`.